app/models/manageiq/providers/openshift/container_manager/refresh_parser.rb

```diff
@@ -47,6 +65,82 @@ def parse_route(route)
+        end
+      end
+
+      def parse_git_source(git_source)
+        parse_strings(git_source,
+                      [:uri, :ref, :httpproxy, :httpsproxy])
+      end
+
+      def parse_build_source(source_item)
+        {
+          :build_source_type  => source_item.try(:type),
+          :source_binary      => source_item.try(:binary).try(:asFile),
+          :source_dockerfile  => source_item.try(:dockerfile),
+          :source_git         => parse_git_source(source_item.try(:git))[:uri],
+          :source_context_dir => source_item.try(:contextDir),
+          :source_secret      => source_item.try(:secret).try(:name)
+        }
+      end
+
+      def parse_object_ref(obj_ref)
+        parse_strings(obj_ref,
+                      [:kind, :namespace, :name,
+                       :uid, :apiVersion, :resourceVersion,
+                       :fieldPath]
+                     )
+      end
+
+      def parse_build_output(output_item)
+        {
+          :to          => parse_object_ref(output_item.try(:to)),
+          :push_secret => output_item.try(:pushSecret).try(:name),
+        }
+      end
+
+      def parse_build(build)
+        new_result = parse_base_item(build)
+        new_result.merge! parse_build_source(build.spec.source)
+        new_result.merge!(
+          :labels                      => parse_labels(build),
+          :service_account             => build.spec.serviceAccount,
+          :completion_deadline_seconds => build.spec.try(:completionDeadlineSeconds),
+        )
+
+        new_result[:output_name] = parse_build_output(build.spec.try(:output))[:to][:name]
+        new_result[:project] = @data_index.fetch_path(:container_projects, :by_name,
+                                                      build.metadata["table"][:namespace])
+        new_result
+      end
+
+      def parse_build_status(build_status)
+        parse_strings(build_status,
+                      [:phase, :reason, :message, :OutputDockerImageReference,
+                       :startTimestamp, :duration, :completionTimestamp])
+      end
+
+      def parse_build_pod(build_pod)
+        new_result = parse_base_item(build_pod)
+        status = parse_build_status(build_pod.status)
+        new_result.merge!(
+          :labels                        => parse_labels(build_pod),
+          :message                       => status[:message],
+          :phase                         => status[:phase],
+          :reason                        => status[:reason],
+          :duration                      => status[:duration],
+          :completion_timestamp          => status[:completionTimestamp],
+          :start_timestamp               => status[:startTimestamp],
+          :output_docker_image_reference => status[:OutputDockerImageReference],
+        )
+        new_result[:build_config] = @data_index.fetch_path(:container_builds, :by_name,
+                                                           build_pod.status.config.try(:name))
```

Comment:
Yes, but it will be OK: fetch_path will not find a key nil, will return nil and this is the expected result.